src/date-range-picker/calendar/grids/index.tsx

```diff
@@ -98,11 +107,17 @@ export const Grids = ({
     event.preventDefault();
 
     handleKey(event, {
-      onActivate: () => focusedDate && onSelectDate(focusedDate),
```

Comment:
Previously focused date was considered as enabled, which is not a case after disabled reasons introduction